src/blockdata/script.rs

```diff
@@ -268,13 +271,21 @@ impl Script {
     }
 
     /// Generates P2WPKH-type of scriptPubkey
-    pub fn new_v0_wpkh(pubkey_hash: &WPubkeyHash) -> Script {
-        Script::new_witness_program(WitnessVersion::V0, &pubkey_hash.to_vec())
+    pub fn new_wpkh(pubkey_hash: &WPubkeyHash) -> Script {
+        Script::new_witness_program(WitnessVersion::V0, &pubkey_hash[..])
     }
 
     /// Generates P2WSH-type of scriptPubkey with a given hash of the redeem script
-    pub fn new_v0_wsh(script_hash: &WScriptHash) -> Script {
-        Script::new_witness_program(WitnessVersion::V0, &script_hash.to_vec())
+    pub fn new_wsh(script_hash: &WScriptHash) -> Script {
+        Script::new_witness_program(WitnessVersion::V0, &script_hash[..])
+    }
+
+    /// Generates P2TR-type of scriptPubkey from a provided public key value,
+    /// tweaking the key with its own TapTweak-tagged hash according to BIP-341
+    pub fn new_p2tr(ctx: &Secp256k1<Verification>, mut pubkey: schnorr::PublicKey) -> Script {
```

Comment:
Can we rename `pubkey` to `internal_key`?

Reply:
Will do, good point